awx/main/constants.py

```diff
@@ -14,7 +16,18 @@
     'STANDARD_INVENTORY_UPDATE_ENV',
 ]
 
-CLOUD_PROVIDERS = ('azure_rm', 'ec2', 'gce', 'vmware', 'openstack', 'rhv', 'satellite6', 'controller', 'insights', 'terraform', 'openshift_virtualization')
+CLOUD_PROVIDERS = []
+
+
+def load_plugin_names():
```

Comment:
Instead of exposing a `CLOUD_PROVIDERS` constant, import this function into all the places that use it and just call it. To make it computed once, use `@cache` from `functools` (it's there since Python 3.9). Keep calling this function from the `ready()` hook to populate the cache at the right moment. And return the list (or maybe a tuple, as it was before?) instead of mutating external variables.

Reply:
I'd also rename the function to `discover_available_plugin_names()`.

Reply:
Oh, and maybe move the function out of `constants` because it's not a constant.

Reply:
Thank you so much for al of these tips and suggestions!! That makes total sense that it doesn't really belong in `constants` at this point. Do you have an alternative suggestion for where this should live?

Reply:
I'm not very familiar with the project structure, but it seems like making a `main/utils/plugins.py` for this might be a good idea. Just make sure to put in module+function docstrings there to clearly state what the scope of the new module is.

Reply:
I went ahead and renamed the function to `discover_available_plugin_names()` in this commit: https://github.com/ansible/awx/pull/15537/commits/519b3b6afa8bc862bd6e29f9f1aeb79a88a12a76

Reply:
to avoid potential confusion, should we add the type of plugin to the function name rather than just `discover_available_plugin_names()`? this name implies discovering _all_ plugins when in reality it isn't.

I am going to be implementing something similar for the inventory sources so was thinking maybe `discover_cloud_plugin_names()` and maybe `discover_inventory_plugin_names()` for my work to follow a similar pattern?

Reply:
Yeah, I wasn't fully sure what it detects. It does make sense to be more precise in naming.

Reply:
I believe I have addressed all of these comments so I am going to mark this conversation as resolved.

Reply:
@djyasin the constant-like variable is still exposed and caching is home-made. Looks like you addressed the naming and relocation comments, though.